### DISCUSSION TITLE
About fonts, do I need to generate training dataset for each font and train together or can train folder by folder for each font? steve8000818@gmail.com

### DISCUSSION THREAD

**Discussion by nissansz** [buliasz/tesstrain-windows-gui#11] [Q&A] [answered]
About fonts, do I need to generate training dataset for each font and train together or can train folder by folder for each font? steve8000818@gmail.com

**Comment by buliasz** [comment, marked as answer]
It's your choice. If you want to create a language (`.traineddata` file) that will work with different fonts - then do all in one folder. If you want a language file that will work best with just a specific font, you'd create a separate Ground Truth directory for each.

**Comment by nissansz** [comment]
How many samples do I need to create for each character.
For example, 4000 characters, each line contains 10 random characters.
Then 4000*500/10=200,000 is enough?

**Comment by buliasz** [comment]
There's no direct answer for this. I'm not sure how many would be necessary for your text. You can try finding answer in the official Tesseract documentation: https://tesseract-ocr.github.io/tessdoc/
You can also ask on the [main tesstrain repository](https://github.com/tesseract-ocr/tesstrain) as this fork is only GUI related.

**Comment by nissansz** [comment]
Ok. Will try training. Thank you.
